frontend/src/components/dashboard/create/sms/SMSRecipients.tsx

```diff
@@ -43,162 +43,189 @@ const SMSRecipients = ({
     <>
-      <StepSection>
-        <StepHeader
-          title="Upload recipient list in CSV format"
-          subtitle="Step 2"
-        >
-          <p>
-            Only CSV format files are allowed. If you have an Excel file, please
-            convert it by going to File &gt; Save As &gt; CSV (Comma delimited).
-          </p>
-          <p>
-            CSV file must include a <b>recipient</b> column with
-            recipients&apos; mobile numbers
-          </p>
-        </StepHeader>
-
-        {!csvFilename && (
-          <WarningBlock title={'We do not remove duplicate recipients'}>
-            <OutboundLink
-              className={styles.warningHelpLink}
-              eventLabel={i18n._(LINKS.guideRemoveDuplicatesUrl)}
-              to={i18n._(LINKS.guideRemoveDuplicatesUrl)}
-              target="_blank"
+      {!isDemo && (
+        <StepSection>
+          <StepHeader title="Add recipient list" subtitle="Step 2">
+            <p>
+              You may add your recipient list either by uploading a CSV file or
+              inserting a Vault link. Only one option is allowed.
+            </p>
+          </StepHeader>
+
+          <div className={styles.recipientTypeSelector}>
+            <PrimaryButton
+              onClick={() => setRecipientListType(RecipientListType.Csv)}
+              className={cx({
+                [styles.active]: recipientListType === RecipientListType.Csv,
+              })}
             >
-              Learn how to remove duplicates in your excel from our guide.
-            </OutboundLink>
-          </WarningBlock>
-        )}
-
-        <CsvUpload
-          isCsvProcessing={isCsvProcessing}
-          csvInfo={csvInfo}
-          onErrorClose={clearCsvStatus}
-        >
-          <FileInput isProcessing={isUploading} onFileSelected={uploadFile} />
-          <p>or</p>
-          <SampleCsv
-            params={params}
-            defaultRecipient="81234567"
-            setErrorMsg={setErrorMessage}
-          />
-        </CsvUpload>
-        {isDemo && (
-          <InfoBlock title="Limited to 20 recipients">
-            <span>
-              You can only send out to 20 recipients per demo campaign. Only the
-              first 20 rows in your CSV file will be taken.
-            </span>
-          </InfoBlock>
-        )}
-        <ErrorBlock>{errorMessage}</ErrorBlock>
+              Use CSV<i className={cx('bx', 'bx-spreadsheet')}></i>
+            </PrimaryButton>
+            <PrimaryButton
+              onClick={() => setRecipientListType(RecipientListType.Vault)}
+              className={cx({
+                [styles.active]: recipientListType === RecipientListType.Vault,
+              })}
+            >
+              Use Vault link<i className={cx('bx', 'bx-link')}></i>
+            </PrimaryButton>
+          </div>
+        </StepSection>
+      )}
+
+      <StepSection>
+        {renderUploadInput()}
+        <ErrorBlock>{error || sampleCsvError}</ErrorBlock>
```

Comment:
Is this ErrorBlock to be common between csv and url upload, it seems like maybe it belongs to the csv upload block? Since url upload has its own ErrorBlock component?

Reply:
`ErrorBlock` is shared across both CSV and URL upload. It's slightly confusing but the main gist is that the `ErrorBlock` within `CsvUpload` and `UrlUpload` displays the `csvError` while this `ErrorBlock` display API call error (e.g. network error, etc).